evaluate.sql

```diff
@@ -0,0 +1,249 @@
+    from timescaledb_information.jobs where application_name like 'Retention Policy%'
+union all
+    select
+        'compression' as feature,
+        count(*) > 0 as uses_feature
+    from timescaledb_information.compression_settings
+union all
+    select
+        'background_jobs' as feature,
+        count(*) > 0 as uses_feature
+    from timescaledb_information.jobs where job_id >= 1000
+) a
+where
+    uses_feature \gset
+\echo 'TimescaleDB features:' :coalesce
+
+select coalesce(json_agg(json_build_object(extname, extversion)), '[]'::json) agg FROM pg_extension
+where extname not in (
+    'bloom', 'btree_gin', 'btree_gist', 'citext', 'cube', 'dict_int', 'dict_xsyn', 'fuzzystrmatch',
+    'hstore', 'intarray', 'isn', 'lo', 'ltree', 'pg_stat_statements', 'pg_trgm', 'pgcrypto', 'pgpcre',
+    'pgrouting', 'pgstattuple', 'pgvector', 'pg_buffercache', 'plpgsql', 'postgis', 'postgis_raster', 'postgis_sfcgal',
+    'postgis_tiger_geocoder', 'postgis_topology', 'seg', 'tablefunc', 'tcn', 'timescaledb_toolkit',
+    'timescaledb', 'tsm_system_rows', 'tsm_system_time', 'unaccent', 'uuid-ossp') \gset
+\echo 'Unsupported extensions in Timescale Cloud:' :agg
+
+select exists(select 1 from information_schema.columns where is_generated = 'ALWAYS') \gset
+\echo 'Do tables have generated columns:' :exists
+
+select exists (
+    select 1 from pg_stats where
+        schemaname not in (
+            '_timescaledb_internal', '_timescaledb_config', '_timescaledb_catalog', '_timescaledb_cache',
+            'timescaledb_experimental', 'timescaledb_information', '_timescaledb_functions',
+            'information_schema', 'pg_catalog')
+    and
+        (
+            exists (
+                select 1 from unnest(most_common_vals::text::text[]) as v
+                where
+                    v IN ('NaN', 'Infinity', '-Infinity')
+            )
+        or
+            exists (
+                select 1 from unnest(histogram_bounds::text::text[]) as h
+                where
+                    h IN ('NaN', 'Infinity', '-Infinity')
+            )
+        )
+) \gset
+\echo 'Do tables attributes have NaN, Infinity or -Infinity*:' :exists
+
+
+-- Rate of inserts, updates, deletes and transactions (per sec)
+create temp table _mig_eval_t (
+    n int, n_tup_ins numeric, n_tup_upd numeric, n_tup_del numeric, xact_commit numeric
+);
+
+insert into _mig_eval_t
+    select 1, sum(n_tup_ins) n_tup_ins, sum(n_tup_upd) n_tup_upd, sum(n_tup_del) n_tup_del, d.xact_commit
+    from pg_stat_user_tables u join pg_stat_database d on true
+    where
+        u.relname not in ('_mig_eval_t') AND
+        u.schemaname not in (
+            '_timescaledb_config', '_timescaledb_catalog', '_timescaledb_cache',
+            'timescaledb_experimental', 'timescaledb_information', '_timescaledb_functions',
+            'information_schema', 'pg_catalog') AND
+        d.datname = current_database()
+    group by d.xact_commit;
+
+select pg_sleep(:wait_duration) \gset
```

Comment:
Can you use the following trick to define default value for wait_duration? This way, use can avoid unless they want.

```
\if :{?wait_duration}
\else
\set wait_duration 60
\endif
```